common/src/main/java/com/wynntils/services/itemfilter/ItemFilterFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import com.wynntils.utils.type.ErrorOr;
+
+public interface ItemFilterFactory {
+    /**
+     * Creates a new item filter from the given input string.
+     *
+     * @param inputString the input string
+     * @return the created item filter, or a translated error string if the input string is invalid
+     */
+    ErrorOr<? extends ItemFilter> create(String inputString);
+
+    /**
+     * @return the keyword used to identify this filter
+     */
+    String getKeyword();
```

Comment:
keyword. Isn't it just the name of the filter? Perhaps that is better. Maybe even this could be an list. Like "lvl" and "level" are both fully acceptable.